src/override/malloc.cc

```diff
@@ -116,6 +116,65 @@ extern "C"
   }
 #endif
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+  SNMALLOC_EXPORT int
+    SNMALLOC_NAME_MANGLE(reallocarr)(void* ptr, size_t nmemb, size_t size)
+  {
+    int err = errno, r;
+    auto& a = ThreadAlloc::get();
+    bool overflow = false;
+    size_t sz = bits::umul(size, nmemb, overflow);
+    if (sz == 0)
+    {
+      errno = err;
+      return 0;
+    }
+    if (overflow)
+    {
+      errno = err;
+      return EOVERFLOW;
+    }
+
+    if (ptr == nullptr)
+    {
+      ptr = a.alloc(sz);
+      if (SNMALLOC_LIKELY(ptr != nullptr))
+      {
+        errno = err;
+        r = 0;
+      }
+      else
+      {
+        r = ENOMEM;
+      }
+      return r;
+    }
+
+    void* p = a.alloc(sz);
```

Comment:
Sorry this is still really confusing me.  You are allocating to `sz` objects here.  One directly, and then a second with realloc?  Why do we need two?  It looks like this code is leaking memory as 
```C++
void* p = a.alloc(sz);
```
appears to be neither returned nor deallocated.

Reply:
the tests did not return 'not empty allocator' exception (realloc on success ?). ptr is to be modified in the last moment too.